libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -32,43 +34,64 @@ export class DefaultConfigurationSettingsConverter implements ConfigurationSetti
   public Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<string | FeatureFlagValue | SecretReferenceValue>[] {
+  ): SetConfigurationSettingParam<string | SecretReferenceValue>[] {
     let configurationSettings = new Array<
-      SetConfigurationSettingParam<string | FeatureFlagValue>
+      SetConfigurationSettingParam
     >();
-    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
-    let foundFeatureManagement = false;
-    let featureFlags: any = {};
-    let featureManagementKeyWord = "";
-    let enabledForKeyWord = "";
+
+    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam>();
+    let foundMsFmSchema = false;
+    let foundLegacySchema = false;
+    let msFmFeatureManagementKeyWord = "";
+    let legacySchemaFeatureManagementKeyWord = "";
+    let legacySchemaEnabledForKeyWord = "";
+    let legachSchemaRequirementTypeKeyWord = "";
+
+    const featureFlagsDict: any = {};
 
     if (options.format == ConfigurationFormat.Properties && !options.skipFeatureFlags) {
       this.checkFeatureManagementExist(config);
     }
 
     for (let i = 0; i < Constants.FeatureManagementKeyWords.length; i++) {
       if (Constants.FeatureManagementKeyWords[i] in config) {
-        if (foundFeatureManagement) {
-          throw new ArgumentError(
-            "Unable to proceed because data contains multiple sections corresponding to Feature Management."
-          );
+        if (Constants.FeatureManagementKeyWords[i] == Constants.FeatureManagementKeyWords[2]) { //feature_management
```

Comment:
This is not so clear to me, how do think we do something like this:
Adjust the sequence of items in FeatureManagementKeyWords:
```
FeatureManagementKeyWords: [
    "FeatureManagement",
    "featureManagement",
    "feature-management"
    "feature_management"
  ]
```


Then we do check in this way
```
for (let i = 0; i < Constants.FeatureManagementKeyWords.length-1; i++) {
     if (Constants.FeatureManagementKeyWords[i] in config) {
        if (foundLegacyFmSechma ) {
             throw;
        }
        
        // do things.....
       
        foundLegacyFmSechma = true
}

if (Constants.FeatureManagementKeyWords[2] in config) {
         if (foundLegacyFmSechma ){
               // check if there's only "feature_flags" section, if not throw     
         }
         else {
                // process new schema and old schema seperately
         }
}
```

Reply:
This makes sense to me. I have updated it accordingly.

Reply:
@RichardChen820 I may have missed it, but why are we not checking/skipping `"feature_management"` ?